src/headerssync.cpp

```diff
@@ -23,15 +27,34 @@ constexpr size_t REDOWNLOAD_BUFFER_SIZE{14827}; // 14827/624 = ~23.8 commitments
 // re-calculate parameters if we compress further)
 static_assert(sizeof(CompressedHeader) == 48);
 
+static size_t ComputeHeadersCacheMax(const CBlockIndex* chain_start, const std::optional<size_t> cache_bytes, size_t element_size)
+{
+    if (cache_bytes) {
+        return *cache_bytes / element_size;
+    }
+
+    return std::min<size_t>(
+        // 1.1 - Account for 10% more blocks than one every 10min.
+        1.1f * ((NodeClock::now() - NodeSeconds{std::chrono::seconds{chain_start->GetMedianTimePast()}}) / 10min),
+        std::max(
+            // Lower limit
+            REDOWNLOAD_BUFFER_SIZE,
+            // Upper limit - We try to make sure not to use more than 10% of
+            // available RAM for the headers cache.
+            (GetFreeRAM().value_or(0) / 10) / element_size));
```

Comment:
again, this is the nack part for me, too much complexity too early - we could simplify by setting overwritable defaults at first

Reply:
(if you insist on memory-storage, but even in that case I'd prefer at least some investigation for why memory was chosen)